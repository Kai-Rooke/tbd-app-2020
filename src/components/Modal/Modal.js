import React, { useState } from 'react';
import PropTypes from 'prop-types';
import './Modal.css';

const Modal = ({id, btnText, btnClass, btnId, ...props}) => {

    const[show, setShow] = useState(false);

    if(!show) {
        return(
            <button id={btnId} class="btnClass" onClick={() => setShow(true)}>{btnText}</button>
        );
    } else {
        return (
            <>
            <button id={btnId} class="btnClass" onClick={() => setShow(true)}>{btnText}</button>
            <div 
            className="main-modal" 
            id={id}>
                <span className="close" onClick={() => setShow(false)}>&times;</span>
                <div className="main-modal-content">
                    {props.children}
                </div>
            </div>
            <div className="modal-overlay" onClick={() => setShow(false)}></div>
            </>
        );   
    }
}

Modal.propTypes = {
    id: PropTypes.string,
    btnText: PropTypes.string,
    btnClass: PropTypes.string,
    btnId: PropTypes.string
};

Modal.defaultProps = {
    id: null,
    btnText: '',
    btnClass: null,
    btnId: null
};

/* EXAMPLE USE 

<Modal id="hello" btnText="This is a modal button">
    <div>
        <h2>This is a modal</h2>
        <p>Example modal</p>
    </div>
</Modal>

*/

export default Modal;