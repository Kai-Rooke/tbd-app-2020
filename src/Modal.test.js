import React from 'react';
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom";
import Modal from './Modal'

it("renders", () => {
    const { asFragment } = render(<Modal id="hello" btnText="This is a modal button"/>);
    expect(asFragment()).toMatchSnapshot();
});