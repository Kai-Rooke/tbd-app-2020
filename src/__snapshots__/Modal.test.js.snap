// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <button
    class="btnClass"
  >
    This is a modal button
  </button>
</DocumentFragment>
`;
